components/status_dropdown/status_dropdown.jsx

```diff
@@ -116,11 +144,54 @@ export default class StatusDropdown extends React.PureComponent {
         const setAway = needsConfirm ? () => this.showStatusChangeConfirmation('away') : this.setAway;
         const setOffline = needsConfirm ? () => this.showStatusChangeConfirmation('offline') : this.setOffline;
 
+        const times = [
+            {
+                text: '30 mins',
+                val: 'dnd30m',
+                val2: 30,
+            },
+            {
+                text: '1 hour',
+                val: 'dnd1h',
+                val2: 60,
+            },
+            {
+                text: '2 hours',
+                val: 'dnd2h',
+                val2: 120,
+            },
+            {
+                text: 'Today',
+                val: 'dndToday',
+                val2: 1000,
+            },
+            {
+                text: 'Tomorrow',
+                val: 'dndTomorrow',
+                val2: 1000,
+            },
+            {
+                text: 'Custom',
+                val: 'dndCustom',
+                val2: 1000,
+            },
+        ];
+
+        const dndSubMenuItems = times.map((tm) => {
+            return {
+                id: `dndTime-${tm.val}`,
+                direction: 'right',
+                text: tm.text,
+                action: this.setStatusDND(tm),
```

Comment:
You are setting the action here to the result of calling `setStatusDND` instead of setting the action to a function that do that, probably you can solve this just using `() => this.setStatusDND(tm)`. Anyway you are passing the whole `tm` object to the `setStatusDND` when you are using only the `val2` attribute, probably makes sense to pass the `val2` attribute only instead. And by the way, val1 and val2 are awful variable names, probably you should go for sometihng else or even remove the `val` variable if is not strictly neded, you can use the position in the list as key, because this is not going to be modified in runtime.

Reply:
Thanks for the help @jespino. I agree val1 and val2 are awful names but this tm object is not final. I was just trying to get the feature working. I will be changing the representation once I get the idea of implementation details. I pushed it here to get the help.